sql/analyzer/assign_indexes.go

```diff
@@ -42,8 +42,17 @@ func assignIndexes(a *Analyzer, node sql.Node) (map[string]*indexLookup, error)
 			return true
 		}
 
+		aliases := make(map[string]sql.Expression)
+		if prj, ok := filter.Child.(*plan.Project); ok {
```

Comment:
There are two cases that I think we're not handling here but can happen:

- The alias was not defined in this project but in another one. Example:
```
Filter(a = 5)
|- Project(a, mytable.B) <- a here is still an alias
    |- Project(mytable.A as a, mytable.B) <-- it's in another project
```
- Group Bys can also define aliases because they act like projects

Reply:
Are aliases local per query? 
How about shadowing?
What if I have 
```sql
SELECT a as foo from (
    SELECT b as foo from T
)
```

Reply:
Subqueries use their own indexes

Reply:
But how about aliases? Here I have `foo` for a and b.

Reply:
`foo` projected from the subquery should have `T` as table, so it's not an alias. Besides, you stop trying to find an alias for foo as soon as you find one.

Reply:
I added a recursive function to _deeply inspect_ all aliases.
I think group by is not really a case here because we don't use indexes for group by.